File: api/verify.js
export default function handler(req, res) {
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method Not Allowed' });
  }

  let body = '';

  req.on('data', chunk => {
    body += chunk.toString();
  });

  req.on('end', () => {
    try {
      const { username, password } = JSON.parse(body);

      if (
        username === process.env.ADMIN_USERNAME &&
        password === process.env.ADMIN_PASSWORD
      ) {
        return res.status(200).json({ success: true });
      } else {
        return res.status(401).json({ error: 'Invalid credentials' });
      }
    } catch (err) {
      return res.status(400).json({ error: 'Invalid request body' });
    }
  });
}

File: api/verify.py
from flask import Flask, request, jsonify
import os

app = Flask(__name__)

@app.route('/api/verify', methods=['POST'])
def verify():
    data = request.json
    username_input = data.get('username')
    password_input = data.get('password')

    current_username = os.getenv('USERNAME')
    current_password = os.getenv('PASSWORD')

    if current_username == username_input and current_password == password_input:
        return jsonify({'success': True}), 200
    else:
        return jsonify({'error': 'Kesalahan terjadi di username atau password'}), 401

# For Vercel Python Runtime:
# Use 'app' as your entrypoint

# The handler that Vercel looks for:
handler = app
